pdr_backend/lake/test/test_required_etl.py

```diff
@@ -0,0 +1,289 @@
+)
+def test_etl_tables(_sample_etl):
+    _, db, gql_tables = _sample_etl
+
+    # Assert all dfs are not the same size as mock data
+    pdr_predictions_df = db.query_data("SELECT * FROM pdr_predictions")
+    pdr_payouts_df = db.query_data("SELECT * FROM pdr_payouts")
+    
+    # Assert len of all dfs
+    assert len(pdr_predictions_df) == 2057
+    assert len(pdr_payouts_df) == 1870
+
+
+# pylint: disable=too-many-statements
+@enforce_types
+@pytest.mark.parametrize(
+    "_sample_etl", [("2024-05-05", "2024-05-06")], indirect=True
+)
+def test_etl_do_bronze_step(_sample_etl):
+    etl, db, gql_tables = _sample_etl
+
+    # assert the number of predictions we expect to see in prod table
+    # assert all predictions have null payouts
+    table_name = NamedTable.from_dataclass(Prediction).fullname
+    pdr_predictions = db.query_data(
+        "SELECT * FROM {}".format(table_name)
+    )
+    assert len(pdr_predictions) == 2057
+    assert pdr_predictions['payout'].is_null().sum() == 2057
+
+    # assert we have valid payouts to join with predictions
+    table_name = NamedTable.from_dataclass(Payout).fullname
+    pdr_payouts = db.query_data(
+        "SELECT * FROM {}".format(table_name)
+    )
+    assert len(pdr_payouts) == 1870
+    assert pdr_payouts['payout'].is_null().sum() == 0
+    
+    # Work 1: Do bronze
+    etl.do_bronze_step()
+    
+    # assert bronze_pdr_predictios and related tables are created correctly
+    temp_table_name = TempTable.from_dataclass(BronzePrediction).fullname
+    temp_bronze_pdr_predictions_records = db.query_data(
+        "SELECT * FROM {}".format(temp_table_name)
+    )
+    null_payouts = temp_bronze_pdr_predictions_records['payout'].is_null().sum()
+    valid_payouts = temp_bronze_pdr_predictions_records['payout'].is_not_null().sum()
+    
+    # assert temp_bronze_pdr_predictions table that will be moved to production
+    assert null_payouts == 377
+    assert valid_payouts == 1678
+    assert null_payouts + valid_payouts == 2055
+    
+    # move tables to production
+    etl._move_from_temp_tables_to_live()
+    
+    # assert bronze_pdr_predictions_df is created correctly
+    table_name = NamedTable.from_dataclass(BronzePrediction).fullname
+    bronze_pdr_predictions_records = db.query_data(
+        "SELECT * FROM {}".format(table_name)
+    )
+    assert bronze_pdr_predictions_records is not None
+    
+    # verify final production table
+    prod_null_payouts = bronze_pdr_predictions_records['payout'].is_null().sum()
+    prod_valid_payouts = bronze_pdr_predictions_records['payout'].is_not_null().sum()
+
+    assert prod_null_payouts == 377
+    assert prod_valid_payouts == 1678
+    assert prod_null_payouts + prod_valid_payouts == 2055
+
+
+# pylint: disable=too-many-statements
+@enforce_types
+@pytest.mark.parametrize(
+    "_sample_etl", [("2024-05-05_00:00", "2024-05-05_00:40")], indirect=True
+)
+def test_etl_do_incremental_bronze_step(_sample_etl):
```

Comment:
test_etl_do_incremental_bronze_step is now showing all work, and completing correctly e2e
1Step-All Records run == same rows as == 3Step-incremental run